### PR TITLE
[Sage-408] Hero - Foundations Update

### PR DESCRIPTION
**Pull request by anechol**
## Description

Updates Hero component to Foundations design specs:
- Updates row gap to `16px`
- Updates margin for title and body to `36px`
- Updates title and body text
  - Title: Heading 4 styles
  - Body: `Grey 700`
- Adds `borderless` prop
- Adds `custom_background_color` prop

[Figma](https://www.figma.com/file/sMbtLUHSt2vfKgKjyQ3052/Sage?node-id=4760%3A22499)

### Implementation Updates
N/A

### Follow-up Items
[Fix component max width](https://kajabi.atlassian.net/browse/SAGE-450)

## Screenshots

|  Before  |  After  |
|--------|--------|
|||
Screen Shot 2022-04-18 at 4 35 52 PM|Screen Shot 2022-04-18 at 4 35 08 PM

## Testing in `sage-lib`

- View [Hero](http://localhost:4000/pages/component/hero)
- Check that component matches Foundations design specs

## Testing in `kajabi-products`

(LOW) Updates Hero component to Foundations design specs.

## Related

[Sage-408](https://kajabi.atlassian.net/browse/SAGE-408)

### CODE REVIEW

**Review by QuintonJason — APPROVED** [review]
Good to go. Minor update around variable but does not stop this merge

File: packages/sage-assets/lib/stylesheets/components/_hero.scss [hunk 1]
```diff
@@ -55,21 +63,22 @@ $-hero-play-icon-background-color-hover: rgba(sage-color(white, 100), 1);
...
   @media (min-width: sage-breakpoint(md-min)) {
-    margin-top: sage-spacing(md);
+    margin-top: rem(36px);
   }
 }
 
 .sage-hero__body {
   grid-area: content;
-  margin-bottom: sage-spacing(md);
+  margin-bottom: rem(36px);
```
Comment (QuintonJason) [inline packages/sage-assets/lib/stylesheets/components/_hero.scss:80]: Can these dupes become a variable?

File: packages/sage-assets/lib/stylesheets/components/_hero.scss [hunk 2]
```diff
@@ -55,21 +63,22 @@ $-hero-play-icon-background-color-hover: rgba(sage-color(white, 100), 1);
...
+  @extend %t-sage-heading-4;
   grid-area: title;
 
   @media (max-width: sage-breakpoint(sm-max)) {
     padding: 0 sage-spacing(md);
   }
 
   @media (min-width: sage-breakpoint(md-min)) {
-    margin-top: sage-spacing(md);
+    margin-top: rem(36px);
```
Comment (monicawheeler) [inline packages/sage-assets/lib/stylesheets/components/_hero.scss:74]: Would it be possible to add this, `rem(36px)`, as a variable?

File: packages/sage-assets/lib/stylesheets/components/_hero.scss (hunk 1 above)
Comment (monicawheeler) [inline packages/sage-assets/lib/stylesheets/components/_hero.scss:80]: As noted, would it be possible to add this, `rem(36px)`, as a variable?

**Review by monicawheeler — CHANGES_REQUESTED** [review]
Very minor suggestion

**Comment by anechol** [comment]
The Container sizes update seems to have caused a sizing issue with this component's max-width. I've updated the sizing to "full" for now but will open up a follow-up ticket to address this further.
Screen Shot 2022-04-18 at 5 35 29 PM

**Review by monicawheeler — APPROVED** [review]
👍🏻

File: packages/sage-assets/lib/stylesheets/components/_hero.scss [hunk 3]
```diff
@@ -17,12 +18,12 @@ $-hero-play-icon-background-color-hover: rgba(sage-color(white, 100), 1);
 .sage-hero {
   display: grid;
   grid-column-gap: sage-spacing(md);
-  grid-row-gap: sage-spacing(md);
+  grid-row-gap: sage-spacing(sm);
   overflow: hidden;
```
Comment (monkeypox8) [inline packages/sage-assets/lib/stylesheets/components/_hero.scss:22]: Small visual bug here caused by the increased `border-radius` + `overflow: hidden`. The focus ring on the embedded link is cut off at the corners:
image-link-focus

**Review by monkeypox8 — APPROVED** [review]
👍🏼 LGTM. Noted a small visual issue, but not a blocker to merge.

**Pull request merged**